### PR TITLE
Makes Research Cell Controls tgui + fixes PoS research cell

### LINKED ISSUE #12822: containment cell controlls on pos not functional
### Description

see title

### Test Merges

Round ID: [20827](https://scrubby.melonmesa.com/round/20827)
BYOND Version: 515.1604
Server revision compiled on: 
Master commit: [3389ff4e4ab922491a45a301e43b24f4801b1cdc](https://github.com/tgstation/TerraGov-Marine-Corps//commit/3389ff4e4ab922491a45a301e43b24f4801b1cdc)
The following pull requests are currently test merged:
[#12648: 'Hivelord's new ability: Recycler Redux' by novaepee at commit 767df6fa85](https://github.com/tgstation/TerraGov-Marine-Corps//pull/12648)
[#12595: 'Mounted AC rework' by Pariah919 at commit 4452489500](https://github.com/tgstation/TerraGov-Marine-Corps//pull/12595)
[#12735: 'Smokening' by Lumipharon at commit 36daaf9b91](https://github.com/tgstation/TerraGov-Marine-Corps//pull/12735)
[#12580: 'Adds SG-62 Target Rifle, a new SG gun with an underbarreled spotting rifle.' by Pariah919 at commit f31cb292b0](https://github.com/tgstation/TerraGov-Marine-Corps//pull/12580)

### Reproduction Steps

walk into research 
aproach containment control panel 
fail to interact
sad researcher noises

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

makes door display controls tgui + makes the code better somewhat
2023-05-08 15_41_08-Change Me

also fixes PoS cell and flasher
## Why It's Good For The Game

looks better
fixes #12822

## Changelog
:cl:
fix: fixed PoS research cell
add: cell controls are now tgui
/:cl:

### CODE REVIEW

File: code/game/objects/machinery/door_display/door_display.dm [hunk 1]
```diff
@@ -13,99 +13,102 @@
...
+			toggle_doors(state)
+			return TRUE
+		if("shutter")
+			var/state = text2num(params["state"])
+			if(!isnum(state))
+				return FALSE
+			toggle_shutters(state)
+			return TRUE
+		if("flash")
+			for(var/obj/machinery/flasher/flasher in targets)
```
Comment (TiviPlus) [inline code/game/objects/machinery/door_display/door_display.dm:99]: why is this typechecking

File: code/game/objects/machinery/door_display/door_display.dm [hunk 2]
```diff
@@ -125,122 +128,18 @@
...
-	if(!open_shutter)
-		data += "[span_linkoff("Open Door")]<br/>"
-	else
-		if (open)
-			data += "<a href='?src=\ref[src];open=0'>Close Door</a><br/>"
+	has_flash = TRUE
+	has_shutters = TRUE
+
+/obj/machinery/door_display/research_cell/toggle_doors(state)
+	for(var/obj/machinery/door/airlock/door in targets)
```
Comment (TiviPlus) [inline code/game/objects/machinery/door_display/door_display.dm:135]: ditto

File: code/game/objects/machinery/door_display/door_display.dm [hunk 3]
```diff
@@ -13,99 +13,102 @@
 	desc = "A remote control for a door."
 	anchored = TRUE
 	density = FALSE
+	interaction_flags = INTERACT_MACHINE_TGUI
+	///Does this type allow for flashers?
+	var/has_flash = FALSE
```
Comment (TiviPlus) [inline code/game/objects/machinery/door_display/door_display.dm:18]: Why add this if its unused

File: code/game/objects/machinery/door_display/door_display.dm [hunk 4]
```diff
@@ -13,99 +13,102 @@
...
+	///Does this type allow for shutters?
+	var/has_shutters = FALSE
```
Comment (TiviPlus) [inline code/game/objects/machinery/door_display/door_display.dm:20]: Ditto

File: code/game/objects/machinery/door_display/door_display.dm [hunk 5]
```diff
@@ -13,99 +13,102 @@
...
-		if(D.density)
+/obj/machinery/door_display/proc/toggle_doors(state)
+	for(var/obj/machinery/door/door in targets)
+		if(state) //true for open
+			door.open()
+		else
+			door.close()
+	open = state
+
+/obj/machinery/door_display/proc/toggle_shutters(state)
```
Comment (TiviPlus) [inline code/game/objects/machinery/door_display/door_display.dm:61]: Autodocs for procs missing

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/door_display/door_display.dm (hunk 1 above)
Comment (mc-oofert) [reply code/game/objects/machinery/door_display/door_display.dm:99]: targets is a list of all machines linked to it for some reason, remnant of the old code

File: code/game/objects/machinery/door_display/door_display.dm (hunk 3 above)
Comment (mc-oofert) [reply code/game/objects/machinery/door_display/door_display.dm:18]: it is used

Comment (TiviPlus) [reply code/game/objects/machinery/door_display/door_display.dm:18]: 7 references and only one I see it set is here

**Pull request closed without merging**